doc/glossary.rst

```diff
@@ -1351,6 +1351,22 @@ Methods
         If the estimator was not already :term:`fitted`, calling this method
         should raise a :class:`exceptions.NotFittedError`.
 
+        ``transform`` does not preserve the memory order of the input array::
```

Comment:
As mentioned in https://github.com/scikit-learn/scikit-learn/pull/12213#discussion_r221445438, it should be made clear that 'order' means memory layout, not the order of the samples.

Also I'm not sure if we really want an example in the glossary. If we do though, no need for the `print` statements, just e.g. 

```
>>> X.flags.c_contiguous
True
```

would suffice

Reply:
@NicolasHug So do you suggest removing the example or leaving it?

Reply:
@NicolasHug I have removed the example from the glossary.

Reply:
You can leave it for now and wait for a core dev review